src/solana/signing_client.rs

```diff
@@ -0,0 +1,180 @@
+        let mut transaction = Transaction::new_unsigned(message);
+        transaction.sign(&[keypair], recent_blockhash);
+        
+        Ok(transaction)
+    }
+    
+    /// Send instructions as a transaction
+    async fn send_instructions(&self, instructions: Vec<Instruction>) -> anyhow::Result<TransactionResponse> {
+        let transaction = self.create_and_sign_transaction(instructions).await?;
+        let signature = self.send_transaction(&transaction).await?;
+        
+        // Poll for confirmation
+        let confirmed = self.poll_for_signature_confirmation(&signature, 30).await?;
+        
+        if !confirmed {
+            return Err(anyhow::anyhow!("Transaction not confirmed"));
+        }
+        
+        // Get the transaction details for response
+        let slot = self.get_slot().await?;
+        
+        Ok(TransactionResponse {
+            hash: signature.to_string(),
+            success: confirmed,
+            block_height: slot,
+            gas_used: 0, // Solana doesn't have gas, we could use compute units if needed
+        })
+    }
+    
+    /// Transfer SOL to another account
+    async fn transfer_sol(&self, to: &str, amount_lamports: u64) -> anyhow::Result<TransactionResponse> {
+        let keypair = self.get_keypair();
+        let to_pubkey = Pubkey::from_str(to)?;
+        
+        let instruction = solana_sdk::system_instruction::transfer(
+            &keypair.pubkey(),
+            &to_pubkey,
+            amount_lamports,
+        );
+        
+        self.send_instructions(vec![instruction]).await
+    }
+    
+    /// Get SOL balance in lamports
+    async fn get_sol_balance(&self) -> anyhow::Result<u64> {
+        let pubkey = self.get_pubkey();
+        self.get_balance(&pubkey).await
+    }
+    
+    /// Get SOL balance for a specific address
+    async fn get_sol_balance_for_address(&self, address: &str) -> anyhow::Result<u64> {
+        let pubkey = Pubkey::from_str(address)?;
+        self.get_balance(&pubkey).await
+    }
+    
+    /// Airdrop SOL (only works on test networks)
+    async fn airdrop_sol(&self, amount_lamports: u64) -> anyhow::Result<TransactionResponse> {
+        let rpc_client = self.get_rpc_client();
+        let pubkey = self.get_pubkey();
+        
+        let signature = rpc_client.request_airdrop(&pubkey, amount_lamports).await?;
+        
+        // Poll for confirmation
+        let confirmed = self.poll_for_signature_confirmation(&signature, 30).await?;
+        
+        if !confirmed {
+            return Err(anyhow::anyhow!("Airdrop not confirmed"));
+        }
+        
+        let slot = self.get_slot().await?;
+        
+        Ok(TransactionResponse {
+            hash: signature.to_string(),
+            success: confirmed,
+            block_height: slot,
+            gas_used: 0,
+        })
+    }
+}
+
+/// Implementation of signing client that wraps a keypair
+pub struct SolanaSigningClientImpl {
```

Comment:
imo as `SolanaSigningClientImpl` implements `SolanaSigningClient`, `SolanaRpcClient`, and more methods on itself (` impl SolanaSigningClientImpl {` below), this could just be `SolanaClient` ?